lib/ansible/plugins/lookup/password.py

```diff
@@ -269,30 +272,54 @@ def _write_password_file(b_path, content):
 class LookupModule(LookupBase):
     def run(self, terms, variables, **kwargs):
         ret = []
-
         for term in terms:
             relpath, params = _parse_parameters(term)
             path = self._loader.path_dwim(relpath)
             b_path = to_bytes(path, errors='surrogate_or_strict')
             chars = _gen_candidate_chars(params['chars'])
 
-            changed = False
-            content = _read_password_file(b_path)
+            content = None
+            writer_process = False
+            b_pathdir = os.path.dirname(b_path)
+            # get a unique lock file name for each password file
+            lockfile_name = to_bytes("%s.ansible_lockfile" % hashlib.md5(b_path).hexdigest())
+            lockfile = os.path.join(b_pathdir, lockfile_name)
+            if not os.path.exists(lockfile) and not os.path.exists(b_path):
+                try:
+                    makedirs_safe(b_pathdir, mode=0o700)
+                    fd = os.open(lockfile, os.O_CREAT | os.O_EXCL)
+                    os.close(fd)
+                    writer_process = True
+                except OSError as e:
+                    if not e.strerror == 'File exists':
```

Comment:
This is a weird combo of comparison operator with a logical negation. You probably meant `if e.strerror != 'File exists':`